In [2]:
import pandas as pd

data = pd.read_csv('data.csv')

In [3]:
data

Admit  Gender Dept  Freq
0   Admitted    Male    A   512
1   Rejected    Male    A   313
2   Admitted  Female    A    89
3   Rejected  Female    A    19
4   Admitted    Male    B   353
5   Rejected    Male    B   207
6   Admitted  Female    B    17
7   Rejected  Female    B     8
8   Admitted    Male    C   120
9   Rejected    Male    C   205
10  Admitted  Female    C   202
11  Rejected  Female    C   391
12  Admitted    Male    D   138
13  Rejected    Male    D   279
14  Admitted  Female    D   131
15  Rejected  Female    D   244
16  Admitted    Male    E    53
17  Rejected    Male    E   138
18  Admitted  Female    E    94
19  Rejected  Female    E   299
20  Admitted    Male    F    22
21  Rejected    Male    F   351
22  Admitted  Female    F    24
23  Rejected  Female    F   317

In [4]:
# Number of male and female candidates applied. In total.
# https://stackoverflow.com/questions/28236305/how-do-i-sum-values-in-a-column-that-match-a-given-condition-using-pandas
gender_group = data.groupby('Gender')['Freq'].sum()

In [5]:
gender_group

Gender
Female    1835
Male      2691
Name: Freq, dtype: int64

In [6]:
# using loc
print("Male: ", data.loc[data['Gender'] == 'Male', 'Freq'].sum())
print("Female: ", data.loc[data['Gender'] == 'Female', 'Freq'].sum())

Male:  2691
Female:  1835


In [9]:
# how many rows of male and how many rows of female.
data['Gender'].value_counts()

Male      12
Female    12
Name: Gender, dtype: int64

In [29]:
# Find male and female rejected
gender_admit_group = data.groupby(['Gender', 'Admit'])['Freq'].sum()

In [30]:
gender_admit_group

Gender  Admit   
Female  Admitted     557
        Rejected    1278
Male    Admitted    1198
        Rejected    1493
Name: Freq, dtype: int64

In [35]:
# percentage of women admitted from applications
print("Women admitted percentage among women applicants: ",
      (gender_admit_group['Female']['Admitted'] / gender_group['Female']) *100)
print("Men admitted percentage among men applicants: ",
      (gender_admit_group['Male']['Admitted'] / gender_group['Male']) *100)

Women admitted percentage among women applicants:  30.3542234332
Men admitted percentage among men applicants:  44.5187662579


In [38]:
# number of applicants by gender per Department
gender_dept = data.groupby(['Gender', 'Dept'])['Freq'].sum()
gender_dept

Gender  Dept
Female  A       108
        B        25
        C       593
        D       375
        E       393
        F       341
Male    A       825
        B       560
        C       325
        D       417
        E       191
        F       373
Name: Freq, dtype: int64

In [68]:
data_total = data.join(gender_dept, on=['Gender', 'Dept'], rsuffix='_total')
data_total.head(7)

Admit  Gender Dept  Freq  Freq_total
0  Admitted    Male    A   512         825
1  Rejected    Male    A   313         825
2  Admitted  Female    A    89         108
3  Rejected  Female    A    19         108
4  Admitted    Male    B   353         560
5  Rejected    Male    B   207         560
6  Admitted  Female    B    17          25

In [70]:
data_total['Percentage'] = data_total['Freq']/data_total['Freq_total']
data_total.head(7)

Admit  Gender Dept  Freq  Freq_total  Percentage
0  Admitted    Male    A   512         825    0.620606
1  Rejected    Male    A   313         825    0.379394
2  Admitted  Female    A    89         108    0.824074
3  Rejected  Female    A    19         108    0.175926
4  Admitted    Male    B   353         560    0.630357
5  Rejected    Male    B   207         560    0.369643
6  Admitted  Female    B    17          25    0.680000

In [80]:
data_total.loc[data_total['Admit'] == 'Admitted', ['Percentage', 'Gender', 'Dept']]

Percentage  Gender Dept
0     0.620606    Male    A
2     0.824074  Female    A
4     0.630357    Male    B
6     0.680000  Female    B
8     0.369231    Male    C
10    0.340641  Female    C
12    0.330935    Male    D
14    0.349333  Female    D
16    0.277487    Male    E
18    0.239186  Female    E
20    0.058981    Male    F
22    0.070381  Female    F

Women admitted percentage among women applicants:  30.3542234332

Men admitted percentage among men applicants:  44.5187662579

In 4/6 Depts more percentage of women was placed than men, compared to the request pool of the same gender.
So more women where places from a small group of requestor.

[learning](https://stackoverflow.com/questions/31796973/pandas-dataframe-combining-one-columns-values-with-same-index-into-list)